### ISSUE TITLE
Same Github Action works on push but not on pull_request_review

### ISSUE THREAD

**Issue by spring1843** [aws-actions/configure-aws-credentials#443]
I have an action that runs on both `push` into main and `pull_request_review`.

The problem I'm seeing is that `configure-aws-credentials` works  on push events but fails when triggered by  pull_request_review with the message `Error: Credentials could not be loaded, please check your action inputs: Could not load credentials from any providers`.

I have also seen it not working with:
```
on:
  push:
    tags: ['v*']
```

I'm not seeing any specific errors anywhere, on the AWS side I see:
`2022-05-13T21:40:01Z  7a625f11-6f05-4634-976d-b5a5cc9f62ff  sts.amazonaws.com  AssumeRoleWithWebIdentity  AccessDenied  An unknown error occurred`

Here's what the Github Action looks like:

```
name: TestAction
on:
  push:
    branches: [ main ]
  pull_request_review:
    types: [ submitted ]
permissions:
  id-token: write
  contents: write
jobs:
  testaction:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: aws-actions/configure-aws-credentials@v1
        with:
          role-to-assume: arn:aws:iam::***********:role/Github
          aws-region: us-east-1
```

The trust relationship for the role looks like:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Federated": "arn:aws:iam::**********:oidc-provider/token.actions.githubusercontent.com"
            },
            "Action": "sts:AssumeRoleWithWebIdentity",
            "Condition": {
                "StringEquals": {
                    "token.actions.githubusercontent.com:aud": "sts.amazonaws.com"
                },
                "StringLike": {
                    "token.actions.githubusercontent.com:sub": "repo:Org/Repo:*"
                }
            }
        }
    ]
}
```

**Comment by kellertk** [comment]
Hi there @spring1843, I tested this out in a private repo and it's working for me. Here's my configuration:

```
name: TestAction
on:
  push:
    branches: [ main ]
  pull_request_review:
    types: [ submitted ]
permissions:
  id-token: write
  contents: write
jobs:
  testaction:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: aws-actions/configure-aws-credentials@v1.6.1
        with:
          role-to-assume: arn:aws:iam::***********:role/testOIDC-Role-ZKGEP6WFYGSY	
          aws-region: us-west-2
      - run: env
```

```
{
    "Version": "2008-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Federated": "arn:aws:iam::**********:oidc-provider/token.actions.githubusercontent.com"
            },
            "Action": "sts:AssumeRoleWithWebIdentity",
            "Condition": {
                "StringLike": {
                    "token.actions.githubusercontent.com:sub": "repo:kellertk/test-configure-aws-credentials:*"
                }
            }
        }
    ]
}
```

I tested this on both `push` and `pull_request_review: [ submitted ]` events and got back my credentials as environment variables properly. A couple of things I noticed:

* The `@v1` tag on aws-actions/configure-aws-credentials is quite old. The latest as of now is `@v1.6.1`. Version 1.6.0 was a big change, and one of the things was #284. This alone could be the cause of your issue.
* You have both a `StringEquals` and `StringLike` condition in your trust relationship. The [current documented CloudFormation template](https://github.com/aws-actions/configure-aws-credentials#sample-iam-role-cloudformation-template) only has the `StringLike` condition. Then again, I tested adding both conditions in the trust policy and it still worked fine.
* Make sure that your `StringLike` condition is correct - If `repo:Org/Repo:*` isn't a obfuscation it should be your actual org and repo :)

**Comment by kellertk** [comment]
Some more info: `Error: Credentials could not be loaded, please check your action inputs:` appears as a result of the validateCredentials function, which is only called if we're running in a self-hosted runner (`GITHUB_ACTIONS=true` is not in the environment), or if you're *not* using the GH OIDC provider (`role-to-assume` is set, `process.env.ACTIONS_ID_TOKEN_REQUEST_TOKEN` is in the environment, `access-key-id` is NOT set, and `web-identity-token-file` is NOT set). Here's a pseudo-code way to explain this:

```
if env['GITHUB_ACTIONS'] is not true:
  validateCredentials()
else
  if role-to-asume is not undefined AND env['ACTIONS_ID_TOKEN_REQUEST_TOKEN'] is not undefined AND aws-access-key-id is undefined AND web-identity-token-file is undefined:
    validateCredentials()
```

Basically you shouldn't be hitting the code that's outputting this error if your environment is what we expect.

**Comment by kellertk** [comment]
After further digging, it looks like GitHub simply doesn't populate the environment variables necessary to do OIDC authentication if the GitHub actions token running the job doesn't have the correct permissions to write to the JWT. [GitHub's documentation](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers#adding-permissions-settings) says that in order to perform OIDC, the `id-token: write` permissions must be set - and note that even in the default "permissive" action setting, this permission is set to none. You have to explicitly set `id-token: write` for OIDC to work.

Note that this also means that OIDC cannot work on forks. The `id-token` is artificially limited to `read` permissions on forks [regardless of the configuration](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token).
